src/Microsoft.OData.Core/ODataMetadataJsonOutputContext.cs

```diff
@@ -156,19 +180,55 @@ protected override void Dispose(bool disposing)
             {
                 if (this.jsonWriter != null)
                 {
-                    this.jsonWriter.Flush();
-                    this.jsonWriter.Dispose();
+                    if (this.asynchronousOutputStream != null)
+                    {
+                        this.asynchronousOutputStream.FlushSync();

```

Comment:
>FlushSync [](start = 54, length = 9)

Why are we calling FlushSync, rather than FlushAsync, on the asynchronous stream?